### PR TITLE
Add `environment` command

### PR DESCRIPTION
**Pull request by vernig**
Add `infra:environment` commands: 

* `infra:environment:set`: set the environment (i.e. Pulumi stack)
* `infra:environment:new`: create a new environment (i.e. Pulumi stack) for the projected selected in the Twilio CLI
* `infra:environment:get`: get the current environment (i.e. Pulumi stack)

### CODE REVIEW

**Comment by vernig** [comment]
Tested with: 

* Deploy
  * Run `twilio infra:deploy` 
  * Check that `.twilio-infra` is added 
* Attempt to deploy to the wrong environment
  * After deployment change twilio CLI profile to a different project. 
  * Run `twilio infra:deploy`
  * Should get an error saying that the environment is already associated to a different account
* Destroy 
  * Go back to the original Twilio project 
  * Run `twilio infra:destroy`
  * Check that `.twilio-infra` is removed
* Set a non existing environemnt
  * Run `twilio infra:environment:set fsdhjkhsdkjadh`
  * Should return `Environment fsdhjkhsdkjadh is not defined`
* Add a new environment to an existing deployment
  * Make sure the project selected in the CLI has an entry in `.twilio-infra`
  * Run `twilio infra:environment:new <name>`
  * Should get n error `Your Twilio project `ACxxxx` is already associated to another environment (xxx)`
* Add a new environment
  * Login to a new project in the CLI 
  * Run `twilio infra:environment:new prod`
  * Should succesfully create a new environment. Check that the eenvironment is added to `.twilio-infra` with `deployed = false`
 * Add a new environment to a non deployed project 
   * Run `twilio infra:environment:new prod2`
   * CLI should warn you that you are overwriting the environment associated to the project and prompt to continue

**Comment by vernig** [comment]
@bermudezmt @kaiquelupo I started implementing the environments. The only test I performed so far are the one in the previous comment. Please feel free to add new comment(s) with other test cases you would like to implement.

File: src/commands/infra/environment/new.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        Printer.print(
+          `Your Twilio project ${accountSid} is already associated to another environment (${deploymentEnvironments[accountSid].environment})\n\nUse\n\n  $ twilio profiles:use <profile name>\n\nto select a different project, or:\n\n  $ twilio profiles:create\n\nto create a new profile.`
+        );
+      } else {
+        runPulumiCommand(['stack', 'init', args.environmentName]);
+        if (this.twilioClient) {
+          addInfra(accountSid, getPulumiStack());
+        }
+        Printer.printSuccess(
+          'New environment created. When you are ready, use\n\n  $ twilio infra:deploy\n\nto deploy it to your Twilio prpject'
```
Comment (bermudezmt) [inline src/commands/infra/environment/new.js]:
```suggestion
          'New environment created. When you are ready, use\n\n  $ twilio infra:deploy\n\nto deploy it to your Twilio project'
```

File: src/commands/infra/destroy.js [hunk 2]
```diff
@@ -1,21 +1,16 @@
...
     await runPulumiCommand(['destroy'], true, this.twilioClient);
     try {
-      fs.access('./.twilio-deploy', fs.constants.F_OK, error => {
-        if (!error) {
-          fs.unlinkSync('./.twilio-deploy', { force: true });
-          Printer.printSuccess('Resource(s) destroyed succesfully!');
-        }
-      });
+      removeInfra(this.twilioClient.accountSid);
+      Printer.printSuccess('Resource(s) destroyed succesfully!');
```
Comment (bermudezmt) [inline src/commands/infra/destroy.js]:
```suggestion
      Printer.printSuccess('Resource(s) destroyed successfully!');
```

File: src/commands/infra/deploy.js [hunk 3]
```diff
@@ -1,22 +1,29 @@
...
+} = require('../../utils');
 
 class InfraDeploy extends TwilioClientCommand {
   async run() {
     await super.run();
-    await runPulumiCommand(['up'], true, this.twilioClient);
+    let deployment = getEnvironmentDeployment();
+    if (deployment && deployment !== this.twilioClient.accountSid) {
+      throw new TwilioCliError(
+        `The current stack is already deployed to ${deployment}. Please switch to that profile or define a new envirnoment`
```
Comment (bermudezmt) [inline src/commands/infra/deploy.js]:
```suggestion
        `The current stack is already deployed to ${deployment}. Please switch to that profile or define a new environment`
```

**Review by bermudezmt — CHANGES_REQUESTED** [review]
Added some commit suggestions. 

Spellcheck, @vernig 😜

File: src/commands/infra/destroy.js (hunk 2 above)
Comment (vernig) [reply src/commands/infra/destroy.js]: Thanks @bermudezmt !

File: src/commands/infra/deploy.js (hunk 3 above)
Comment (vernig) [reply src/commands/infra/deploy.js]: Thanks @bermudezmt !

**Comment by vernig** [comment]
@kaiquelupo @bermudezmt the last commit bd36e4b adds the check before creating a new environment. You can test this using the following step: 

* Add a new environment to a non deployed project
  * Create a new IaC project or use an existing one that has not been deployed
  * Run twilio infra:environment:new <new_environment_name> 
  * CLI should warn you that you are overwriting the environment associated to the project and prompt to continue

```
The current Twilio project (ACxxxxx) is already associated to environment "xxxx", but it's not deployed. 
Are you sure you want to associate a new environment to it? Y/n
````
I updated the [comment above](https://github.com/twilio-infra-as-code/plugin-twilio-infra/pull/17#issuecomment-823469048) with the new test.

**Comment by bermudezmt** [comment]
Works beautifully, thank you!

**Pull request merged**